### PR TITLE
Add /rerank and /import sections to README

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

In anticipation of the upcoming major release, we need to update the Go README to incorporate sections for `rerank` and `import`.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [x] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1368,11 +1440,67 @@ Send text to Pinecone's inference API to generate embeddings for documents and q
...
+**Note:** The `score` represents the absolute measure of relevance of a given query and passage pair. Normalized
+between [0, 1], the `score` represents how closely relevant a specific item and query are, with scores closer to 1
+indicating higher relevance.
+
+```go
+    ctx := context.Background()
+
+    pc, err := pinecone.NewClient(pinecone.NewClientParams{
+        ApiKey: "YOUR-API-KEY"
+	})
```
Comment (austin-denoble) [inline README.md:1459]: nit: Indentation may be off here.

**Review by austin-denoble — APPROVED** [review]
LGTM, thanks for adding!

File: README.md [hunk 2]
```diff
@@ -579,6 +579,74 @@ func main() {
...
+
+    clientParams := pinecone.NewClientParams{
+        ApiKey: os.Getenv("PINECONE_API_KEY"),
+    }
+
+    pc, err := pinecone.NewClient(clientParams)
+
+    if err != nil {
+        log.Fatalf("Failed to create Client: %v", err)
+    } else {
```
[the remark this answers is not in the document]
Comment (aulorbe) [reply README.md:605]: okie doke

File: README.md [hunk 3]
```diff
@@ -579,6 +579,74 @@ func main() {
...
+    idxConnection, err := pc.Index(pinecone.NewIndexConnParams{Host: idx.Host})
+    if err != nil {
+        log.Fatalf("Failed to create IndexConnection for Host: %v: %v", idx.Host, err)
+    }
+
+    storageURI := "s3://my-bucket/my-directory/"
+
+    errorMode := "abort" // Will abort if error encountered; other option: "continue"
+	
+    importRes, err := idxConnection.StartImport(ctx, storageURI, nil, (*pinecone.ImportErrorMode)(&errorMode))
```
[the remark this answers is not in the document]
Comment (aulorbe) [reply README.md:640]: sounds good!

File: README.md (hunk 1 above)
Comment (aulorbe) [reply README.md:1459]: It looks good on my screen; it's aligned with `pc`. What are you seeing?
Screenshot 2024-10-21 at 4 48 22 PM

Comment (austin-denoble) [reply README.md:1455]: I think it's just gh formatting, nevermind.

Comment (aulorbe) [reply README.md:1455]: okay cool!

**Pull request merged**